vnext/Microsoft.ReactNative/Utils/TransformableText.h

```diff
@@ -0,0 +1,55 @@
+#pragma once
+
+#include <boost/algorithm/string.hpp>
+
+namespace react::uwp {
+struct TransformableText final {
+  enum class TextTransform { Undefined, None, Uppercase, Lowercase, Capitalize };
+
+  TextTransform m_textTransform{TextTransform::Undefined};
+  std::wstring m_originalText{};
+
+  std::wstring TransformText() {
+    switch (m_textTransform) {
+      case TextTransform::Undefined:
+        [[fallthrough]];
+      case TextTransform::None:
+        return m_originalText;
+      case TextTransform::Uppercase:
+        return boost::to_upper_copy(m_originalText);
+      case TextTransform::Lowercase:
+        return boost::to_lower_copy(m_originalText);
+      case TextTransform::Capitalize: {
+        auto str = std::make_unique<wchar_t[]>(m_originalText.length() + 1);
+        const int nChars = LCMapStringW(
+            LOCALE_NAME_USER_DEFAULT,
+            LCMAP_TITLECASE,
+            m_originalText.c_str(),
+            static_cast<int>(m_originalText.length()),
+            str.get(),
+            static_cast<int>(m_originalText.length() + 1));
+
+        return std::move(str.get());
```

Comment:
We're doing an extra copy from the c-string buffer to wstring. Would reccomend the alternative solution mentioned in the other comment of creating a wstring, resizing it to the desired text length, then using the non-const ".data()" for C string interop.

Reply:
I think you intended to avoid the copy by doing a move of the raw pointer, but note wstring doesn't do anything special with rvalues to c string pointers https://en.cppreference.com/w/cpp/string/basic_string/basic_string

Doing a move of the raw pointer probably isn't what you want in that case in general though since it won't clear out the unique_ptr. There's a unique_ptr method like "take" or something like that which does though.

Reply:
> We're doing an extra copy from the c-string buffer to wstring. Would reccomend the alternative solution mentioned in the other comment of creating a wstring, resizing it to the desired text length, then using the non-const ".data()" for C string interop with LCMapStringW.

The problem with using data() is that yes, you can write to it but the size member of the `std::string` doesn't change so effectively the string is empty. The solution is to use `resize` instead of `reserve` to ensure the size is set before/after